newIDE/app/src/EventsSheet/InstructionEditor/InstructionOrExpressionSelector/index.js

```diff
@@ -52,178 +52,167 @@ type Props<T> = {|
   onClickMore?: () => void,
   id?: ?string,
 |};
-type State<T> = {|
+type SearchState<T> = {|
   searchText: string,
   searchResults: Array<T>,
 |};
 
-export default class InstructionOrExpressionSelector<
+const InstructionOrExpressionSelector = <
   T: EnumeratedInstructionOrExpressionMetadata
-> extends React.PureComponent<Props<T>, State<T>> {
-  state: State<T> = {
+>({
+  focusOnMount,
+  instructionsInfo,
+  instructionsInfoTree,
+  selectedType,
+  onChoose,
+  iconSize,
+  useSubheaders,
+  searchPlaceholderObjectName,
+  searchPlaceholderIsCondition,
+  helpPagePath,
+  style,
+  onClickMore,
+  id,
+}: Props<T>) => {
+  const searchBarRef = React.useRef<?SearchBarInterface>(null);
+  const scrollViewRef = React.useRef<?ScrollViewInterface>(null);
+  const selectedItemRef = React.useRef<?ListItemRefType>(null);
+  const [searchState, setSearchState] = React.useState<SearchState<T>>({
     searchText: '',
     searchResults: [],
-  };
-  _searchBar: ?SearchBarInterface;
-  _scrollView = React.createRef<ScrollViewInterface>();
-  _selectedItem = React.createRef<ListItemRefType>();
-  searchApi = null;
-
-  initialInstructionTypePath = findInTree(
-    this.props.instructionsInfoTree,
-    this.props.selectedType
+  });
+  const searchApi = React.useMemo(
+    () =>
+      new Fuse(instructionsInfo, {
+        ...sharedFuseConfiguration,
+        keys: [
+          { name: 'displayedName', weight: 2 },
+          { name: 'fullGroupName', weight: 1 },
+        ],
+      }),
+    [instructionsInfo]
+  );
+  const initialInstructionTypePathRef = React.useRef<?(string[])>(
+    findInTree(instructionsInfoTree, selectedType)
   );
 
-  componentDidMount() {
-    if (this._selectedItem.current && this._scrollView.current) {
-      this._scrollView.current.scrollTo(this._selectedItem.current);
-    }
-
-    this.searchApi = new Fuse(this.props.instructionsInfo, {
-      ...sharedFuseConfiguration,
-      keys: [
-        { name: 'displayedName', weight: 2 },
-        { name: 'fullGroupName', weight: 1 },
-      ],
-    });
-  }
-
-  focus = () => {
```

Comment:
wasn't used?

Reply:
I couldn't find somewhere where it was used

Reply:
(No parent component gives it a ref)

Reply:
Maybe it was not removed at the time the focusOnMount props was added as a better replacement